jose.pyx

```diff
@@ -0,0 +1,141 @@
+from libc.stdlib cimport free
+cimport jansson
+cimport jose
+
+import json
+
+def jwk_generate(jwk):
+    cdef jansson.json_t *cjwk = NULL
+    cdef char *ret = NULL
+
+    assert isinstance(jwk, dict)
+
+    try:
+        cjwk = jansson.json_loads(json.dumps(jwk).encode(u"UTF-8"), 0, NULL)
+        assert cjwk
+
+        assert jose.jose_jwk_generate(cjwk)
+
+        ret = jansson.json_dumps(cjwk, 0)
+        assert ret
+
+        return json.loads(ret)
+    finally:
+        jansson.json_decref(cjwk)
+        free(ret)
+
+
+def jwk_clean(jwk):
+    cdef jansson.json_t *cjwk = NULL
+    cdef char *ret = NULL
+
+    assert isinstance(jwk, dict)
+
+    try:
+        cjwk = jansson.json_loads(json.dumps(jwk).encode(u"UTF-8"), 0, NULL)
+        assert cjwk
+
+        assert jose.jose_jwk_clean(cjwk)
+
+        ret = jansson.json_dumps(cjwk, 0)
+        assert ret
+
+        return json.loads(ret)
+    finally:
+        jansson.json_decref(cjwk)
+        free(ret)
+
+
+def jwk_allowed(jwk, req=False, use=None, op=None):
+    cdef jansson.json_t *cjwk = NULL
+    cdef const char *cuse = NULL
+    cdef const char *cop = NULL
+
+    assert isinstance(jwk, dict)
+    assert op is None or isinstance(op, unicode)
+    assert use is None or isinstance(use, unicode)
```

Comment:
You can augment the function signature and let Cython do the error handling and type checking.